lib/YaST/workarounds.pm

```diff
@@ -0,0 +1,51 @@
+# Copyright 2015-2022 SUSE LLC
+# SPDX-License-Identifier: GPL-2.0-or-later
+
+package YaST::workarounds;
+
+
+use strict;
+use warnings;
+use Exporter 'import';
+use testapi;
+use version_utils;
+
+
+our @EXPORT = qw(apply_workaround_bsc1204176);
+
+=head1 Workarounds for known issues
+
+=head2 apply_workaround_bsc1204176 ($needle, [,[$timeout] | timeout => $timeout] ):
+
+Workaround for the screen refresh issue.
+
+First checks if we need to apply the workaround, if the needle matches 
+already no workaround is required.
+
+If the workaround is needed we record a soft failure and apply a 'shift-f3' 
+and 'esc' sequence that should fix the problem
+
+If the problem still persists (we saw one occasion in the VRs) then we retry
+with maximazing and shrinking the screen twice by sending 'alt-f10' two times. 
+
+=cut
+
+sub apply_workaround_bsc1204176 {
```

Comment:
The code from check_screen can be reused, it takes into account 1 or more optional parameters, this is why I think it happens:

```
fn([qw(needle1 needle2)], timeout => 5); # call with named param

my ($needle) = shift; #unpack needle or array ref of needles
my $timeout = shift if (@_ % 2); # @_ contains a list `qw(timeout 5)` so module operation is `0` and `$timeout` is not initialized
my %args = (timeout => $timeout // 0, @_); # %args is (timeout => 0, timeout => 5) where the second key/value overrides the first

fn('needle',5); # call with positional param

my ($needle) = shift; #unpack needle or array ref of needles
my $timeout = shift if (@_ % 2); # the remainder is 1 so `$timeout` is set and `@_` is empty.
my %args = (timeout => $timeout // 0, @_); # $timeout is set and we got %args = (timeout => 5, ()) and the empty list doesn't append anything as it is not a reference.
```

Conclusion: Perl is full of magic!